api/src/middleware/merge-content-versions.ts

```diff
@@ -0,0 +1,36 @@
+import type { RequestHandler } from 'express';
+import { VersionsService } from '../services/versions.js';
+import asyncHandler from '../utils/async-handler.js';
+import { mergeVersionsRaw, mergeVersionSaves } from '../utils/merge-version-saves.js';
+
+/**
+ * Checks and merges
+ */
+export const mergeContentVersions: RequestHandler = asyncHandler(async (req, res, next) => {
```

Comment:
I know its still WIP, just want to leave this comment as reminder for later to update/finish this comment